### PR TITLE
[APR] Add retry clause when fetching tvl/volume

### PR DESCRIPTION
**Pull request by Rawallon**
Fixes days without tvl, following Fabio's recommendation on [BAL-872](https://linear.app/bleu-llc/issue/BAL-872/handle-days-were-snapshot-does-not-exist#comment-be6f5eac)

[On Prod](https://tools.balancer.blue/apr/pool/ethereum/0x1a44e35d5451e0b78621a1b3e7a53dfaa306b1d000000000000000000000051b?startAt=09-14-2023&endAt=09-22-2023)
image

With the fix

image

### CODE REVIEW

File: apps/balancer-tools/src/app/apr/(utils)/calculatePoolStats.ts [hunk 1]
```diff
@@ -53,7 +56,26 @@ const fetchPoolAveragesForDateRange = withCache(
...
+        from: from - SECONDS_IN_DAY * 7,
+        to,
+      });
+
+      if (retyGQL.poolSnapshots.length === 0) {
+        return [0, 0, "", []];
+      }
+      // Gets the 2 most recent
+      res.poolSnapshots = retyGQL.poolSnapshots
+        .slice(-2)
```
Comment (Rawallon) [inline apps/balancer-tools/src/app/apr/(utils)/calculatePoolStats.ts:77]:
```suggestion
      res.poolSnapshots = retyGQL.poolSnapshots
        .slice(-1)
```

File: apps/balancer-tools/src/app/apr/(utils)/calculatePoolStats.ts [hunk 2]
```diff
@@ -53,7 +56,25 @@ const fetchPoolAveragesForDateRange = withCache(
...
+      const retyGQL = await pools.gql(network).poolSnapshotInRange({
+        poolId,
+        from: from - SECONDS_IN_DAY * 7,
+        to,
+      });
+
+      if (retyGQL.poolSnapshots.length === 0) {
+        return [0, 0, "", []];
+      }
+      res.poolSnapshots = retyGQL.poolSnapshots
```
Comment (luizakp) [inline apps/balancer-tools/src/app/apr/(utils)/calculatePoolStats.ts:75]: this get the last data and return. e.g. Monday TVL = 10, Tuesday nothing happened then TVL = 10 too

**Comment by luizakp** [comment]
/apr/pool/ethereum/0x1a44e35d5451e0b78621a1b3e7a53dfaa306b1d000000000000000000000051b?startAt=09-14-2023&endAt=09-22-2023

**Comment by ribeirojose** [comment]
> Fixes days without tvl, following Fabio's recommendation on [BAL-872](https://linear.app/bleu-llc/issue/BAL-872/handle-days-were-snapshot-does-not-exist#comment-be6f5eac)
> 
> [On Prod](https://tools.balancer.blue/apr/pool/ethereum/0x1a44e35d5451e0b78621a1b3e7a53dfaa306b1d000000000000000000000051b?startAt=09-14-2023&endAt=09-22-2023) image
> 
> With the fix
> 
> image

- Since TVL values won't change all that much every day, we could scale the y axis not to start from 0 for better readability. I think we do this in the pool simulator cc @luizakp @yvesfracari;
- IMO the APR chart should be in front of all the other charts (TVL is currently on top of it);
- #would try to add some transparency to the colors in bar charts (similar to what we did in the homepage) and avoid using the same colors in more than one chart to make them look a bit nicer when together / one on top of another;

**Pull request merged**